docs/subnets/create-a-simple-rust-vm.md

```diff
@@ -0,0 +1,1256 @@
+        self.state = state;
+    }
+
+    /// Verifies Block properties (e.g., heights),
+    /// and once verified, records it to the State.
+    pub async fn verify(&mut self) -> io::Result<()> {
+        if self.height == 0 && self.parent_id == ids::Id::empty() {
+            log::debug!(
+                "block {} has an empty parent Id since it's a genesis block -- skipping verify",
+                self.id
+            );
+            self.state.add_verified(&self.clone()).await;
+            return Ok(());
+        }
+
+        // if already exists in database, it means it's already accepted
+        // thus no need to verify once more
+        if self.state.get_block(&self.id).await.is_ok() {
+            log::debug!("block {} already verified", self.id);
+            return Ok(());
+        }
+
+        let prnt_blk = self.state.get_block(&self.parent_id).await?;
+
+        if prnt_blk.height != self.height - 1 {
+            return Err(Error::new(
+                ErrorKind::InvalidData,
+                format!(
+                    "parent block height {} != current block height {} - 1",
+                    prnt_blk.height, self.height
+                ),
+            ));
+        }
+
+        if prnt_blk.timestamp > self.timestamp {
+            return Err(Error::new(
+                ErrorKind::InvalidData,
+                format!(
+                    "parent block timestamp {} > current block timestamp {}",
+                    prnt_blk.timestamp, self.timestamp
+                ),
+            ));
+        }
+
+        self.state.add_verified(&self.clone()).await;
+        Ok(())
+    }
+
+    /// Mark this Block accepted and updates State accordingly.
+    pub async fn accept(&mut self) -> io::Result<()> {
+        self.set_status(choices::status::Status::Accepted);
+
+        // only decided blocks are persistent -- no reorg
+        self.state.write_block(&self.clone()).await?;
+        self.state.set_last_accepted_block(&self.id()).await?;
+
+        self.state.remove_verified(&self.id()).await;
+        Ok(())
+    }
+
+    /// Mark this Block rejected and updates State accordingly.
+    pub async fn reject(&mut self) -> io::Result<()> {
+        self.set_status(choices::status::Status::Rejected);
+
+        // only decided blocks are persistent -- no reorg
+        self.state.write_block(&self.clone()).await?;
+
+        self.state.remove_verified(&self.id()).await;
+        Ok(())
+    }
+}
+```
+
+#### verify
+
+This method verifies that a block is valid and stores it in the memory. It is important to store the verified block in the memory and return them in the `vm.get_block()` method.
+
+```rust title="/timestampvm/src/block/mod.rs"
+/// Verify returns Ok if the block is valid.
+/// Verifies Block properties (e.g., heights),
+/// b.parent.Timestamp < b.Timestamp <= [local time] + 1 hour
+/// and once verified, records to State memory.
+pub async fn verify(&mut self) -> io::Result<()> {
```

Comment:
The comments above no longer matches https://github.com/ava-labs/timestampvm-rs/blob/main/timestampvm/src/block/mod.rs#L171